step-sequencer/src/audio/jack/step_sequencer_client.rs

```diff
@@ -223,18 +223,22 @@ fn process_beatmaker(
 ) -> SSResult<()> {
     // TODO: Can NOT use while loop to process all messages in the channel.
     // Find out why.
-    if let Ok(event) = &subscription.receiver.try_recv() {
-        debug!("BeatMaker: subscription ID: {:?}", subscription.id);
-        debug!("BeatMaker: Received event from: {:?}", event);
-        let data = event.to_data()?;
-        debug!("BeatMaker: MIDI data: {:?}", data);
-        let time = match event {
-            ChannelVoiceEvent::NoteOff { .. } => 100,
-            _ => 0,
-        };
-        let raw_midi = RawMidi { time, bytes: &data };
-        let mut midi_writer = port.writer(process_scope);
-        midi_writer.write(&raw_midi)?;
+    let mut midi_writer = port.writer(process_scope);
```

Comment:
The reason why while was not working before is that we create writer in every loop. The writer maintains its own buffer to write. At the end of every loop, the writer is dropped and in the next loop we overwrite whatever is written by the previous writer.